packages/rs-dpp/src/data_contract/config/mod.rs

```diff
@@ -48,107 +46,73 @@ impl DataContractConfig {
             }),
         }
     }
-
-    /// Retrieve contract configuration properties.
-    ///
-    /// This method takes a BTreeMap representing a contract and retrieves
-    /// the configuration properties based on the values found in the map.
-    ///
-    /// The process of retrieving contract configuration properties is versioned,
-    /// and the version is determined by the platform version parameter.
-    /// If the version is not supported, an error is returned.
-    ///
-    /// # Parameters
-    ///
-    /// * `contract`: BTreeMap representing the contract.
-    /// * `platform_version`: The platform version being used.
-    ///
-    /// # Returns
-    ///
-    /// * `Result<ContractConfig, ProtocolError>`: On success, a ContractConfig.
-    ///   On failure, a ProtocolError.
-    pub(in crate::data_contract) fn get_contract_configuration_properties(
-        contract: &BTreeMap<String, Value>,
-        platform_version: &PlatformVersion,
-    ) -> Result<DataContractConfig, ProtocolError> {
-        match platform_version.dpp.contract_versions.config_version {
-            0 => Ok(
-                DataContractConfigV0::get_contract_configuration_properties_v0(contract)?.into(),
-            ),
-            version => Err(ProtocolError::UnknownVersionMismatch {
-                method: "DataContractConfig::get_contract_configuration_properties".to_string(),
-                known_versions: vec![0],
-                received: version,
-            }),
-        }
-    }
 }
 
 impl DataContractConfigGettersV0 for DataContractConfig {
-    fn can_be_deleted(&self) -> bool {
+    fn is_contract_deletion_allowed(&self) -> bool {
         match self {
-            DataContractConfig::V0(v0) => v0.can_be_deleted,
+            DataContractConfig::V0(v0) => v0.allow_contract_deletion,
         }
     }
 
-    fn readonly(&self) -> bool {
+    fn is_contract_update_allowed(&self) -> bool {
         match self {
-            DataContractConfig::V0(v0) => v0.readonly,
+            DataContractConfig::V0(v0) => v0.allowe_contract_update,
         }
     }
 
-    fn keeps_history(&self) -> bool {
+    fn keep_previous_contract_versions(&self) -> bool {
         match self {
-            DataContractConfig::V0(v0) => v0.keeps_history,
+            DataContractConfig::V0(v0) => v0.keep_previous_contract_versions,
         }
     }
 
-    fn documents_keep_history_contract_default(&self) -> bool {
+    fn document_revisions(&self) -> bool {
```

Comment:
No, let's keep documents_keep_history_contract_default, other ones I can be okay with, this one I can not.